Framework/Crystal/src/FilterPeaks.cpp

```diff
@@ -34,7 +36,10 @@ double SN(const Mantid::Geometry::IPeak &p) { return p.getIntensity() / p.getSig
 
 double RUN(const Mantid::Geometry::IPeak &p) { return p.getRunNumber(); }
 
-std::string BANKNAME(const Mantid::Geometry::IPeak &p) { return p.getBankName(); }
+std::string BANKNAME(const Mantid::Geometry::IPeak &p) {
+  const Peak &fullPeak = dynamic_cast<const Peak &>(p);
+  return fullPeak.getBankName();
```

Comment:
We probably need a check here.

Or actually, maybe we should disable filtering by bank name if it's a LeanElasticPeaksWorkspace.

```python
peaks = CreatePeaksWorkspace(OutputType="LeanElasticPeak", NumberOfPeaks=0)
p = peaks.createPeakQSample([1,1,1])
peaks.addPeak(p)
out=FilterPeaks(peaks, BankName='bank')
```
```
FilterPeaks-[Error] Error in execution of algorithm FilterPeaks:
FilterPeaks-[Error] std::bad_cast
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
~/src/mantid/Framework/PythonInterface/mantid/simpleapi.py in __call__(self, *args, **kwargs)
   1041                 try:
-> 1042                     algm.execute()
   1043                 except RuntimeError as e:

RuntimeError: std::bad_cast
```

Reply:
I'm including additional condition `if (!bankname.empty() && (inputWS->id() == "PeaksWorkspace"))` so that `bankname` is ignored for non `PeakWorkspace` workspaces.